pkg/location/location.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package location
+
+import (
+	"fmt"
+)
+
+type Location struct {
+	Reference     Reference
+	ReferenceLock ReferenceLock
+}
```

Comment:
Having fun with naming here :) ..alternatives for `Reference` and `ReferenceLock` might be `Addr` and `ResolvedAddr` 

```

loc.Addr

loc.ResolvedAddr

```

Reply:
Interesting --- do you mean to change the field names to 
```
Addr Reference
ResolvedAddr ReferenceLock
```
or rename the two 
```
type Reference interface {...}
type ReferenceLock interface {...}
```
as
```
type Addr interface {...}
type ResolvedAddr interface {...}
```
?

I think (see above) the `location.Location{}` struct should probably be deleted because the purpose it serves doesn't deserve such a significant sounding name... So those field names would go away with it...

That said if there's a better name than "Reference" for the core interface, now would be the time to rename it. :) Though I tend to suggest avoiding abbreviations in type names...
```
type Address interface {...}
type ResolvedAddress interface {...}

addr, err := location.ParseAddress(arg, location.WithZzz(), location.WithYyy())
src, err := content.Open(addr)
```
or
```
type Location interface {...}
type ResolvedLocation interface {...}

loc, err := location.ParseLocation(arg, location.WithZzz(), location.WithYyy())
src, err := content.Open(loc)
```

I _kind_ of like "address" thinking just of git and oci... But thinking about `location.Dir{}` and "-" streams... A custom location could even have a `map[string]string` as its only field if you wrote a content-opener and filesystem-provider for it... Point being I do prefer Address initially but I think it breaks down thinking about the non-remote kinds of location/reference.